src/mpid/ch4/include/mpidpre.h

```diff
@@ -227,7 +227,31 @@ typedef struct MPIDI_prequest {
     MPI_Datatype datatype;
 } MPIDI_prequest_t;
 
+/* Structures for partitioned pt2pt request */
+typedef struct MPIDIG_part_sreq {
+    MPIR_cc_t ready_cntr;
+} MPIDIG_part_sreq_t;
+
+typedef struct MPIDIG_part_rreq {
+    MPI_Aint sdata_size;        /* size of entire send data */
+
+    /* fields for posted/unexp list */
+    MPIR_Request *request;      /* pointer of part_req */
+    struct MPIDIG_part_rreq *prev, *next;
+} MPIDIG_part_rreq_t;
+
+typedef struct MPIDIG_part_request {
+    MPL_atomic_int_t status;    /* see MPIDIG_PART_REQ_INC_FETCH_STATUS */
+    MPIR_Request *peer_req_ptr;
+    union {
+        MPIDIG_part_sreq_t sreq;
+        MPIDIG_part_rreq_t rreq;
```

Comment:
In too many places, `sreq` and `rreq` are of type `MPIR_Request *`, so the names here are a source of confusion. I think you can just use `send` and `recv` here.

Reply:
Sure. will replace